src/DocumentSession.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021, Kung-Yao Hsieh, Shang-Chen LIN, Yi-Hang Xie. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef HTTPLIB_SRC_DOCUMENTSESSION_H
+#define HTTPLIB_SRC_DOCUMENTSESSION_H
+
+#include <unordered_set>
+#include <string>
+#include <memory>
+#include <mutex>
+
+class DocumentManager;
+class SocketSession;
+
+/** @brief Manage the operation between SocketSession and DocumentManager. */
+class DocumentSession {
+private:
+    /** @brief Store DocumentManager pointer. It is used to synchronize datas. */
+    DocumentManager* doc_manager_;
+    /** @brief Store SocketSession object pointer. */
+    std::unordered_set<std::shared_ptr<SocketSession>> sessions_;
+    /** @brief Lock sessions_ when it is used. */
+    std::mutex m_sessions_;
+    /** @brief Parse update parameter reveived from client message.
+     *  @param update_params Client message.
+     */
+    void parseUpdateParameters(const std::string& update_params);
+    /** @brief Syncronize the updates to clients and database. */
+    void syncDocument();
+public:
+    DocumentSession();
+    DocumentSession(const DocumentSession&) = delete;
+    /** @brief Remove session after socekt connect.
+     *  @param session SocketSession object that will be added to session list.
+     */
+    void addSession(std::shared_ptr<SocketSession> session);
+    /** @brief Remove session after socekt close.
+     *  @param session SocketSession object that will be removed from session list.
+     */
+    void removeSession(std::shared_ptr<SocketSession> session);
```

Comment:
Please consider if shared_ptr is necessary. This may cause memory leak to be undetected.

Reply:
It is not necessary to use share_ptr, so I use raw pointer as function parameter.